database-commons/src/main/java/io/cdap/plugin/db/source/AbstractDBSource.java

```diff
@@ -270,6 +290,9 @@ public void prepareRun(BatchSourceContext context) throws Exception {
     }
     context.setInput(Input.of(sourceConfig.getReferenceName(), new SourceInputFormatProvider(
       DataDrivenETLDBInputFormat.class, connectionConfigAccessor.getConfiguration())));
+
+    // set error details provider
+    context.setErrorDetailsProvider(new ErrorDetailsProviderSpec(getErrorDetailsProviderClassName()));
```

Comment:
similarly here: 

`context.setErrorDetailsProvider` should be called before `context.setInput(Input.of(sourceConfig.getReferenceName(), new SourceInputFormatProvider(
      DataDrivenETLDBInputFormat.class, connectionConfigAccessor.getConfiguration())))` is called.

Otherwise error details provider information is not propagated to platform.

Reply:
Moved !